x-pack/plugin/rollup/src/main/java/org/elasticsearch/xpack/downsample/RollupShardIndexer.java

```diff
@@ -86,7 +91,15 @@ class RollupShardIndexer {
     private final Rounding.Prepared rounding;
     private final List<FieldValueFetcher> fieldValueFetchers;
     private final RollupShardTask task;
+    /*
```

Comment:
We could move the old BulkProcessor into this package? I think BulkProcessor2 is faster (since it allows for multiple index requests to be in flight at once), but I don't actually have any performance tests (I don't know if there are any for this).

Reply:
What if I add a new `addWithBackpressure` method in BulkProcessor2, and move this code in there? If I understand correctly, your concern is not that we run everything on the current thread (which would be fairly difficult with BulkProcessor2), but that you don't want to get exceptions if the search code is running much faster than the index code, and you don't want the complexity (or the need to maintain it) in TSDB code. Right?

Reply:
> What if I add a new addWithBackpressure method in BulkProcessor2 (a separate method will allow us to pass in a Supplier to check whether RollupShardIndexer.abort is true), and move this code in there? 

👍 this sounds good to me. The code added in this change to `RollupShardSearcher` isn't about rolling up data, but being able to bulk index on the current thread.